### PR TITLE
chore: migrate to ginkgo v2

### LINKED ISSUE #1952: [e2e] Migrate to ginkgo v2
#### What would you like to be added:

Ginkgo v1 has been deprecated, we should migrate to v2. Guide: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#upgrading-to-ginkgo-20

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Ginkgo v1 has been deprecated, we should migrate to v2.

#### Which issue(s) this PR fixes:

Fixes #1952

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -359,7 +359,7 @@ test-e2e-capz: ## Run k8s e2e tests with capz
 	hack/test_k8s_e2e_capz.sh $(TEST_E2E_ARGS)
 
 test-ccm-e2e: ## Run cloud provider e2e tests.
-	go test ./tests/e2e/ -timeout 0 -v -ginkgo.v $(CCM_E2E_ARGS)
+	hack/test-ccm-e2e.sh
```
Comment (lzhecheng) [inline Makefile:362]: Shall we delete definition of `CCM_E2E_ARGS`?

Comment (nilo19) [reply Makefile:362]: Yes, we should. I will delete it if the tests are stable.

**Comment by lzhecheng** [comment]
@nilo19 the basic-lb test fix is merged. Is this PR ready? Then I will approve.

**Comment by nilo19** [comment]
> @nilo19 the basic-lb test fix is merged. Is this PR ready? Then I will approve.

yes, it is.

**Pull request merged**